extendr-api/src/wrapper/matrix.rs

```diff
@@ -220,9 +229,15 @@ where
         self.as_typed_slice_mut().unwrap()
     }
 
-    /// Get the dimensions for this array.
-    pub fn dim(&self) -> &D {
-        &self.dim
+    /// Returns the number of dimensions.
+    pub fn ndim(&self) -> usize {
+        NDIM
+    }
+
+    /// Returns the dimensions of the array.
+    // TODO:: Need to deprecate this and use `get_dim` instead.
```

Comment:
Personally, I'm fine with two methods to get the same piece of data. I don't think it needs deprecation. Other areas have similar inconsistencies...